frontend/src/pages/modelServing/screens/global/useRouteForInferenceService.ts

```diff
@@ -14,6 +16,13 @@ const useRouteForInferenceService = (
   const routeNamespace = inferenceService.metadata.namespace;
 
   React.useEffect(() => {
+    if (isKServe) {
+      const kserveRoute = getUrlFromKserveInferenceService(inferenceService);
+      setRoute(kserveRoute || null);
+      setLoaded(true);
+      setLoadError(kserveRoute ? null : new Error('Route not found'));
+      return;
+    }
```

Comment:
`inferenceService` is a bad object to use as a dependency on `useEffect` as unrelated changes can re-trigger the `useEffect`. Pre-compute the kserve route URL and use it as a dependency. For example:
```suggestion
  const kserveRoute = isKServe ? getUrlFromKserveInferenceService(inferenceService) : null;
  React.useEffect(() => {
    if (isKServe) {
      setRoute(kserveRoute);
      setLoaded(true);
      setLoadError(kserveRoute ? null : new Error('Route not found'));
      return;
    }
```

Reply:
oh right, ugh im sorry, that's a n00b error, thanks!